icu-date.lua

```diff
@@ -50,10 +55,19 @@ local function string_to_uchar(str)
   return uchar
 end
 
-local function uchar_to_string(uchar)
-  local length = call_fn("u_strlen", uchar) + 1
-  local str = ffi.new('char[?]', length)
-  call_fn("u_austrcpy", str, uchar)
+local function uchar_to_string(uchar, length)
+  local str = ffi.new('char[?]', length + 1)
+  local status, needed_length = call_fn_status("ucnv_fromUChars", cnv, str, length, uchar, -1, status_ptr)
```

Comment:
If length is known, why we use `-1` for the source buffer length (it means calling of a strlen-like function)?

Reply:
Hm. I see that when our initial guess about buffer size was okay, we pass this guess (`result_length`) instead of an actual string length (`needed_length`). So just replacing `-1` with the `length` parameter will not work.

Reply:
This is not a blocker (the code looks workable, just a bit non optimal). So I'll push it now and we can discuss optimizations later.

Reply:
BTW, we can pass the known byte buffer length (returned by the `ucnv_fromUChar()`) to `ffi.string()`.